From 21b0095dfd0b84ec1056ad415cbd8b55517d6e7d Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:29 +0800
Subject: [PATCH 3026/4585] [Backport] sw64: add missing global
 __constant_c_memset

Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56U83

Reference: https://gitee.com/openeuler/kernel/commit/eceee39b861763f3ac3e70fe5306641146cfef7f

--------------------------------

This patch fixes undefined reference to __constant_c_memset if
CONFIG_DEEP_MEMSET=y and CONFIG_VGA_CONSOLE=y.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 arch/sw_64/lib/deep-memset.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/sw_64/lib/deep-memset.S b/arch/sw_64/lib/deep-memset.S
index 4efba2062e11..ffadc9a52707 100644
--- a/arch/sw_64/lib/deep-memset.S
+++ b/arch/sw_64/lib/deep-memset.S
@@ -35,6 +35,7 @@
 	.globl __memset
 	.globl ___memset
 	.globl __memsetw
+	.globl __constant_c_memset
 	.ent ___memset
 ___memset:
 	.frame $30, 0, $26, 0
-- 
2.17.1

